api/tests/helpers.ts

```diff
@@ -0,0 +1,200 @@
+import { readdirSync } from 'fs';
+import { PgStore } from '../src/pg/pg-store';
+import { FastifyBaseLogger, FastifyInstance } from 'fastify';
+import { IncomingMessage, Server, ServerResponse } from 'http';
+import { TypeBoxTypeProvider } from '@fastify/type-provider-typebox';
+import { buildApiServer } from '../src/api/init';
+import { Rune } from '../src/api/schemas';
+import { DbLedgerEntry, DbRune } from '../src/pg/types';
+
+export type TestFastifyServer = FastifyInstance<
+  Server,
+  IncomingMessage,
+  ServerResponse,
+  FastifyBaseLogger,
+  TypeBoxTypeProvider
+>;
+
+export async function startTestApiServer(db: PgStore): Promise<TestFastifyServer> {
+  return await buildApiServer({ db });
+}
+
+export async function runMigrations(db: PgStore) {
+  const contents = readdirSync('../migrations');
+  await db.sqlWriteTransaction(async sql => {
+    for (const fileName of contents) {
+      if (!fileName.endsWith('.sql')) continue;
+      await db.sql.file('../migrations/' + fileName);
+    }
+  });
+}
+
+export async function dropDatabase(db: PgStore) {
+  await db.sqlWriteTransaction(async sql => {
+    // Drop all tables.
+    await sql`
+      DO $$ DECLARE
+          r RECORD;
+      BEGIN
+          FOR r IN (SELECT tablename FROM pg_tables WHERE schemaname = current_schema()) LOOP
+              EXECUTE 'DROP TABLE IF EXISTS ' || quote_ident(r.tablename) || ' CASCADE';
+          END LOOP;
+      END $$
+    `;
+    // Drop all types.
+    await sql`
+      DO $$ DECLARE
+          r RECORD;
+      BEGIN
+          FOR r IN (SELECT typname FROM pg_type WHERE typtype = 'e' AND typnamespace = (SELECT oid FROM pg_namespace WHERE nspname = current_schema())) LOOP
+              EXECUTE 'DROP TYPE IF EXISTS ' || quote_ident(r.typname) || ' CASCADE';
+          END LOOP;
+      END $$;
+    `;
+  });
+}
+export function sampleLedgerEntry(rune_id: string, block_height?: string): DbLedgerEntry {
+  return {
+    rune_id: '1:1',
+    block_hash: '0000000000000000000320283a032748cef8227873ff4872689bf23f1cda83a5',
+    block_height: block_height || '840000',
+    tx_index: 0,
+    tx_id: '2bb85f4b004be6da54f766c17c1e855187327112c231ef2ff35ebad0ea67c69e',
+    output: 0,
+    address: '0',
+    receiver_address: '0',
+    amount: '0',
+    operation: 'etching',
+    timestamp: 0,
+  };
+}
+
+function toSpacedName(name: string | null): string | null {
+  if (name === null) {
+    return null;
+  }
+  // should take "Some name" and make it "Some•name"
+  const words = name.split(' ');
+  return words.join('•');
+}
+export function sampleRune(id: string, name?: string): DbRune {
+  return {
+    id: '1:1',
+    name: name || 'SAMPLE RUNE NAME',
```

Comment:
Rune names never have spaces so your default name here should be `SAMPLERUNENAME`